jaeger-extension-native/src/main/java/io/ballerina/observe/trace/jaeger/JaegerTracerProvider.java

```diff
@@ -46,34 +58,58 @@ public void init() {    // Do Nothing
     }
 
     public static void initializeConfigurations(BString agentHostname, int agentPort, BString samplerType,
-                                                  BDecimal samplerParam, int reporterFlushInterval,
-                                                  int reporterBufferSize) {
-        // Create Sampler Configuration
-        samplerConfiguration = new Configuration.SamplerConfiguration()
-                .withType(samplerType.getValue())
-                .withParam(samplerParam.value());
-
-        // Create Sender Configuration
-        Configuration.SenderConfiguration senderConfiguration = new Configuration.SenderConfiguration()
-                .withAgentHost(agentHostname.getValue())
-                .withAgentPort(agentPort);
+                                                BDecimal samplerParam, int reporterFlushInterval,
+                                                int reporterBufferSize) {
+
         String reporterEndpoint = agentHostname + ":" + agentPort;
 
-        // Create Reporter Configuration
-        reporterConfiguration = new Configuration.ReporterConfiguration()
-                .withSender(senderConfiguration)
-                .withFlushInterval(reporterFlushInterval)
-                .withMaxQueueSize(reporterBufferSize);
+        ManagedChannel jaegerChannel = new NettyChannelProvider()
+                .builderForTarget(reporterEndpoint)
+                .usePlaintext()
+                .build();
+
+        OtlpGrpcSpanExporter exporter = OtlpGrpcSpanExporter.builder()
+                .setChannel(jaegerChannel)
+                .build();
+
+        tracerProviderBuilder = SdkTracerProvider.builder()
+                .addSpanProcessor(BatchSpanProcessor
+                        .builder(exporter)
+                        .setMaxExportBatchSize(reporterBufferSize)
+                        .setExporterTimeout(reporterFlushInterval, TimeUnit.MILLISECONDS)
+                        .build());
+
+        switch (samplerType.getValue()) {
+            default:
+            case "const":
+                if (samplerParam.value().intValue() == 0) {
+                    tracerProviderBuilder.setSampler(Sampler.alwaysOff());
+                } else {
+                    tracerProviderBuilder.setSampler(Sampler.alwaysOn());
+                }
+                break;
+            case "probabilistic":
+                tracerProviderBuilder.setSampler(Sampler.traceIdRatioBased(samplerParam.value().doubleValue()));
+                break;
+            case "ratelimiting":
```

Comment:
is it better to use `RateLimitingSampler.TYPE` here?

Reply:
fixed